example/basic.go

```diff
@@ -0,0 +1,63 @@
+package main
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+
+	"github.com/RisingStack/trace-go/trace"
+)
+
+var collector = trace.NewMemoryCollector()
+
+func handle2(rw http.ResponseWriter, r *http.Request) {
+	span := trace.NewSpanIDFromRequest(r)
+	log.Println("Id in 2nd endpoint: " + span.String() + " on URL: " + r.URL.String())
+	resp := struct {
+		Message string `json:"message"`
+	}{
+		"Message from Endpoint2 to Endpoint1",
+	}
+	json.NewEncoder(rw).Encode(resp)
+}
+
+func handle(rw http.ResponseWriter, r *http.Request) {
+	msg := struct {
+		Message string `json:"message"`
+	}{
+		"Message from Endpoint1 to Endpoint2",
+	}
+	span := trace.NewSpanIDFromRequest(r)
+	log.Println("Id in 1st endpoint: " + span.String())
+	c := http.Client{
+		Transport: trace.Transport{
```

Comment:
Can we make this API a little bit nicer? Something like:

```go
c: = http.Client{Transport: trace.Transport(r)}
```

I think we can get the span id behind the scenes and the user only have to get the span id if she wants to use it.